### ISSUE TITLE
Loss in output

### ISSUE THREAD

**Issue by NguyenThanhTan** [sfzhang15/FaceBoxes#34]
While training Faceboxes model, I noticed that the output is strange:
```
I0316 10:16:10.963383  6670 solver.cpp:243] Iteration 30450, loss = 3.36783
I0316 10:16:10.963630  6670 solver.cpp:259]     Train net output #0: mbox_loss = 3.34174 (* 1 = 3.34174 loss)
I0316 10:16:13.457460  6670 sgd_solver.cpp:151] Iteration 30450, lr = 0.0052975
I0316 10:18:44.811120  6670 solver.cpp:243] Iteration 30500, loss = 3.35862
I0316 10:18:44.812209  6670 solver.cpp:259]     Train net output #0: mbox_loss = 4.07279 (* 1 = 4.07279 loss)
I0316 10:18:44.812288  6670 sgd_solver.cpp:151] Iteration 30500, lr = 0.005275
I0316 10:21:14.092057  6670 solver.cpp:243] Iteration 30550, loss = 3.36363
I0316 10:21:14.092276  6670 solver.cpp:259]     Train net output #0: mbox_loss = 3.34809 (* 1 = 3.34809 loss)
I0316 10:21:14.092325  6670 sgd_solver.cpp:151] Iteration 30550, lr = 0.0052525
I0316 10:23:43.217934  6670 solver.cpp:243] Iteration 30600, loss = 3.35409

```
Why isn't there classification loss but only localization loss?

I looked into `solver.cpp`, the code is as below:
```
    if (display) {
      LOG_IF(INFO, Caffe::root_solver()) << "Iteration " << iter_
          << ", loss = " << smoothed_loss_;
      const vector<Blob<Dtype>*>& result = net_->output_blobs();
      int score_index = 0;
      for (int j = 0; j < result.size(); ++j) {
        const Dtype* result_vec = result[j]->cpu_data();
        const string& output_name =
            net_->blob_names()[net_->output_blob_indices()[j]];
        const Dtype loss_weight =
            net_->blob_loss_weights()[net_->output_blob_indices()[j]];
        for (int k = 0; k < result[j]->count(); ++k) {
          ostringstream loss_msg_stream;
          if (loss_weight) {
            loss_msg_stream << " (* " << loss_weight
                            << " = " << loss_weight * result_vec[k] << " loss)";
          }
          LOG_IF(INFO, Caffe::root_solver()) << "    Train net output #"
              << score_index++ << ": " << output_name << " = "
              << result_vec[k] << loss_msg_stream.str();
        }
      }
    }
```
The for loop iterates all the losses and prints them, if so, why isn't there classification loss?

**Comment by NguyenThanhTan** [comment]
I changed `solver.prototxt` so that it displays every iteration and average_loss is set to 1, turn out mbox loss is weighted sum of classification loss and localization loss:
```
I0318 07:00:49.299060  9887 solver.cpp:243] Iteration 2, loss = 22.4237
I0318 07:00:49.299134  9887 solver.cpp:259]     Train net output #0: mbox_loss = 22.4237 (* 1 = 22.4237 loss)
I0318 07:00:51.557700  9887 sgd_solver.cpp:151] Iteration 2, lr = 0.001
I0318 07:00:51.989467  9887 solver.cpp:243] Iteration 3, loss = 18.6429
I0318 07:00:51.989526  9887 solver.cpp:259]     Train net output #0: mbox_loss = 18.6429 (* 1 = 18.6429 loss)
I0318 07:00:54.307365  9887 sgd_solver.cpp:151] Iteration 3, lr = 0.001
I0318 07:00:54.840184  9887 solver.cpp:243] Iteration 4, loss = 17.7905
I0318 07:00:54.840260  9887 solver.cpp:259]     Train net output #0: mbox_loss = 17.7905 (* 1 = 17.7905 loss)
I0318 07:00:57.176961  9887 sgd_solver.cpp:151] Iteration 4, lr = 0.001
I0318 07:00:57.703336  9887 solver.cpp:243] Iteration 5, loss = 16.4505
I0318 07:00:57.703442  9887 solver.cpp:259]     Train net output #0: mbox_loss = 16.4505 (* 1 = 16.4505 loss)
I0318 07:01:00.236745  9887 sgd_solver.cpp:151] Iteration 5, lr = 0.001
```
